vpr/src/route/rr_graph2.h

```diff
@@ -75,40 +68,36 @@ vtr::NdMatrix<int, 2> get_number_track_to_track_inter_die_conn(t_sb_connection_m
                                                                const int custom_3d_sb_fanin_fanout,
                                                                RRGraphBuilder& rr_graph_builder);
 
-t_seg_details* alloc_and_load_seg_details(int* max_chan_width,
-                                          const int max_len,
-                                          const std::vector<t_segment_inf>& segment_inf,
-                                          const bool use_full_seg_groups,
-                                          const enum e_directionality directionality,
-                                          int* num_seg_details = nullptr);
+std::vector<t_seg_details> alloc_and_load_seg_details(int* max_chan_width,
+                                                      const int max_len,
+                                                      const std::vector<t_segment_inf>& segment_inf,
+                                                      const bool use_full_seg_groups,
+                                                      const enum e_directionality directionality);
 
 void alloc_and_load_chan_details(const DeviceGrid& grid,
-                                 const t_chan_width* nodes_per_chan,
-                                 const int num_seg_details_x,
-                                 const int num_seg_details_y,
-                                 const t_seg_details* seg_details_x,
-                                 const t_seg_details* seg_details_y,
+                                 const t_chan_width& nodes_per_chan,
+                                 const std::vector<t_seg_details>& seg_details_x,
+                                 const std::vector<t_seg_details>& seg_details_y,
                                  t_chan_details& chan_details_x,
                                  t_chan_details& chan_details_y);
+
 t_chan_details init_chan_details(const DeviceGrid& grid,
```

Comment:
If there are comments in the .cpp files on what some or all of these do, it would be good to make Doxygen comments here.